ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using ServerCore.DataModel;
+using ServerCore.Models;
+
+namespace ServerCore.ModelBases
+{
+    public abstract class EventSpecificPageModel : PageModel
+    {
+        [FromRoute]
+        [ModelBinder(typeof(EventBinder))]
+        public Event Event { get; set; }
+
+        public class EventBinder : IModelBinder
+        {
+            public async Task BindModelAsync(ModelBindingContext bindingContext)
+            {
+                string eventIdAsString = bindingContext.ActionContext.RouteData.Values["eventId"] as string;
```

Comment:
Should we consider adding logic to translate from a string like "PuzzleHunt2019" to the id here? That would give us friendly urls instead of having to find/translate database ids.

Reply:
Reading my mind here. I want to add another property to Event like "ShortName" and key off that instead of ID. Shortname examples our existing site uses: 19, 2018.5, etc.

I just didn't want to add a database change to this PR as well.